web/src/main/java/co/mobileaction/example/web/service/PostService.java

```diff
@@ -25,6 +27,17 @@ public void savePost(Post post)
         postRepository.save(post);
     }
 
+    @Override
+    public Post convertFrom(PostDto postDto)
+    {
+        return Post.builder()
+                .id(postDto.getId())
+                .userId(postDto.getUserId())
+                .title(postDto.getTitle())
+                .body(postDto.getBody())
+                .build();
+    }

```

Comment:
you can directly use this inside `savePost()` method instead of defining an interface method. Or (the better way), you can define this conversion method inside PostDto object, and use it in the controller like `postService.savePost(postDto.convertToPost());`, it will be more compact. Also, you don't need to get `id` of user from the endpoint, you can define `id` as `serial`, and let the database set the `id`.